app/controllers/video-list.controller.js

```diff
@@ -0,0 +1,69 @@
+(function() {
+  'use strict';
+
+  class VideoListCtrl {
+    constructor(StorageService, VideoService) {
+      Object.assign(this, {StorageService, VideoService});
+      this.StorageService.get('videos').then(videos => {
+        this.videos = JSON.parse(JSON.stringify(videos)) || [];
+      });
+      this.StorageService.get('videos').then(videos => {
+        this.temporaryVideos = JSON.parse(JSON.stringify(videos)) || []; //variable created for button SHOW ALL after filtering this.videos - ONLY FAV
+      });
+      this.options = {
+        pageSizes: [5, 10, 20, 50, 100],
+        currentPageSize: 5,
+        currentPage: 0,
+        display: 'list',
+        orderProp: 'title',
+        onlyFavourites: false
+      }
+    }
+    
+    add(link) {
+      this.VideoService.add(link).then((videoData) => {
+        if (!videoData) {return;}
+        this.videos.push(videoData);
+        this.temporaryVideos = [];
+        this.videos.map(val => {
+          this.temporaryVideos.push(val);
+        });
+        return this.StorageService.set('videos', this.videos);

```

Comment:
chcialbym bardziej, zebys do StorageService nie wysylala za kazdym razem pelnej listy elementow. raczej identyfikator elementu, ktory chcesz usunac i cala operacje, zeby zalatwial storageService. 

dlaczego?
chodzi o przekazywanie duzych zestawow danych. w przypadku zapisu do localStorage nie robi to roznicy. w przypadku gdybys zapisywala je callem do backendu mialabys problem gdyby nagle okazalo sie, ze tablica wazy 2 czy 3 mega.

takze sugeruje zrobienie API storageService na wzor calli restowych:
. add / post
. get / get
. update / put
. remove / delete

(wprowadz prosze poprawki wszedzie)